File: src/leetcode/java/bit_manipulation/MissingNumber_268.java
package leetcode.java.bit_manipulation;

public class MissingNumber_268 {
  public int missingNumber(int[] nums) {
    int xor = 0, i = 0;
    for (; i < nums.length; i++) {
      xor = xor ^ i ^ nums[i];
    }
    return xor ^ i;
  }
}

File: src/leetcode/python/bit_manipulation/MissingNumber_268.py
class MissingNumber_268(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        xor = 0
        i = 0
        for num in nums:
            xor = xor ^ i ^ num
            i += 1
        return xor ^ i
